0.preprocess-sites/8.image-and-segmentation-qc.py

```diff
@@ -0,0 +1,398 @@
+import os
+import sys
+import pathlib
+import argparse
+import pandas as pd
+import matplotlib.pyplot as plt
+import seaborn as sns
+
+sys.path.append(os.path.join("..", "scripts"))
+from config_utils import process_config_file
+from cell_quality_utils import CellQuality
+
+parser = argparse.ArgumentParser()
+parser.add_argument(
+    "--config_file",
+    help="configuration yaml file for preprocessing pipeline",
+    default="site_processing_config.yaml",
+)
+args = parser.parse_args()
+config_file = args.config_file
+
+config = process_config_file(config_file)
+
+# Defines the sections of the config file
+core_args = config["core"]
+spots_args = config["process-spots"]
+summ_cell_args = config["summarize-cells"]
+summ_plate_args = config["summarize-plate"]
+
+# Defines the variables set in the config file
+batch = core_args["batch"]
+quality_func = core_args["categorize_cell_quality"]
+ignore_files = core_args["ignore_files"]
+sites_per_image_grid_side = core_args["sites_per_image_grid_side"]
+batch_dir = core_args["batch_dir"]
+
+cell_filter = spots_args["cell_filter"]
+
+summ_cells_results_basedir = summ_cell_args["output_resultsdir"]
+summ_cells_figures_basedir = summ_cell_args["output_figuresdir"]
+cell_category_order = summ_cell_args["cell_category_order"]
+
+correlation_threshold = summ_plate_args["correlation_threshold"]
+painting_image_names = summ_plate_args["painting_image_names"]
+barcoding_cycles = summ_plate_args["barcoding_cycles"]
+barcoding_prefix = summ_plate_args["barcoding_prefix"]
+
+cell_count_file = pathlib.Path(
+    summ_cells_results_basedir, batch, "cells", "cell_count.tsv"
+)
+cell_count_df = pd.read_csv(cell_count_file, sep="\t")
+
+figures_output = pathlib.Path(summ_cells_figures_basedir, batch)
+os.makedirs(figures_output, exist_ok=True)
+results_output = pathlib.Path(summ_cells_results_basedir, batch)
+os.makedirs(results_output, exist_ok=True)
+
+# Creates x, y coordinates for plotting per-plate views.
+# Assumes image numbering starts in upper left corner and proceeds down
+final_order = []
+for i in range(1, sites_per_image_grid_side + 1):
+    build_seq = list(
+        zip(
+            ([i] * (sites_per_image_grid_side + 1)),
+            reversed(range(1, (sites_per_image_grid_side + 1))),
+        )
+    )
+    final_order = final_order + build_seq
```

Comment:
ooh, you're going to love += assignment for lists

try `final_order += build_seq` (it is the same as above)

Reply:
oh yeah! I forgot that was a thing :) thanks!